src/compiler/Stage/Emit/JavaScript.elm

```diff
@@ -66,6 +66,11 @@ emitExpr ( expr, _ ) =
             in
             "((() => {" ++ bindingsJS ++ "; return " ++ emitExpr body ++ ";})())"
 
+        List { list } ->
+            "["
+            ++ (List.map emitExpr list |> String.join ", ")
+            ++ "]"
```

Comment:
Official elm represents lists as linked lists rather than javascript arrays which makes concatenation work. I suspect that elm-in-elm should follow the same approach

Reply:
Interestingly official Elm keeps list literals as Lists of the language it's defined in (see [AST.Source](https://github.com/elm/compiler/blob/master/compiler/src/AST/Source.hs#L52), [AST.Canonical](https://github.com/elm/compiler/blob/master/compiler/src/AST/Canonical.hs#L91), [AST.Optimized](https://github.com/elm/compiler/blob/master/compiler/src/AST/Optimized.hs#L60)) and when emitting JS ([Generate.JavaScript.Expression](https://github.com/elm/compiler/blob/master/compiler/src/Generate/JavaScript/Expression.hs#L103-L113)) either returns `Nil` or calls `fromArray` and generates a JS list as an argument for it.

We'll definitely have to plug into the `elm/core` List definition (btw funny thing, it's [JS-only](https://github.com/elm/core/blob/1.0.2/src/Elm/Kernel/List.js) :sob: ), but we can do that later. What @Warry has right now is not that far off, and for now good enough I think. When we have enough stuff working that it stops being consistent with each other, we'll fix it.

WDYT @harrysarson @Warry ?

Reply:
That is interesting, also interesting why the list definition is in javascript. 

Regarding long term plan sounds good - the point at which it stops working will probably be when we implement `(::)`.